_posts/2023-06-15-user-survey-2023.md

```diff
@@ -0,0 +1,240 @@
+  </div>
+  <div class="col">
+    <img src="/assets/blog/2023-06-15-user-survey-2023/7-install-python.svg" class="img-fluid mx-auto d-block" alt="Python installation source">
+  </div>
+</div>
+
+cocotb pushes a new release roughly twice a year, and takes great care in ensuring backwards-compatibility to allow our users to switch to the latest version.
+The survey results confirm that this effort is paying off: 83 percent of users are on the latest 1.7 release (at the time of the survey).
+
+<img src="/assets/blog/2023-06-15-user-survey-2023/10-cocotb-version.svg" class="img-fluid mx-auto d-block p-5" alt="cocotb version">
+
+A look at the simulators used with cocotb completes the picture.
+When choosing proprietary simulators, Cadence Xcelium and Siemens ModelSim and Questa lead the field.
+In the world of open source simulators, Verilator and Icarus Verilog are the most popular choices.
+Icarus Verilog is the default simulator in many cocotb tutorials and even its Makefiles, so its popularity is hardly surprising.
+Equally, GHDL is the most mature open source VHDL simulator, hence being the go-to choice for VHDL developers.
+
+What is surprising, however, is the amount of people using Verilator together with cocotb.
+Verilator is an excellent simulator, however, it lacks a number of features that make it integrate nicely with cocotb.
+It seems that these limitations are not severe enough to prevent its use with cocotb for many users.
+
+<img src="/assets/blog/2023-06-15-user-survey-2023/9-simulator.svg" class="img-fluid mx-auto d-block p-5" alt="Simulator">
+
+Finally, when asked for their wishlist of platforms cocotb should support, the most common answers were
+
+* support for the Vivado simulator (xsim) [which unfortunately does not implement the VPI or VHPI interface], and
+* Better support for GHDL [which is partially due to the fact that GHDL is uniquely using the VPI interface for VHDL, as well as missing features in GHDL itself].
+
+## Verification IP or cocotb extensions
+
+**Most cocotb users use cocotb-test and a coverage extension, such as pyvsc or cocotb-coverage. Python implementations of UVM are popular with cocotb users as well, as is (AXI) bus IP.**
+
+Cocotb is intentionally not solving every possible problem: it's extensible and encourages users to choose their preferred extensions from the Python package ecosystem.
+
+For users who need more than cocotb's Makefile-based build system, cocotb-test is the go-to choice.
+Functional coverage and constraint-random test generation are next on the list, with users mostly relying on pyvsc or cocotb-coverage.
+
+As we'll see below, the availability of verification IP (VIP) is one of the weak spots of cocotb.
+Still, high-quality VIP is available, and many users resort to cocotbext-axi for AXI bus adapters or cocotbext-spi for SPI adapters.
+Additionally, cocotb-bus provides a further set of bus interfaces which is commonly used.
```

Comment:
Sure we want to mention cocotb-bus, as it's pretty unmaintained?

Reply:
It still gets 10k downloads a month and is widely used, so it apparently provides some value to its users.